web/src/app/util/PageActions.js

```diff
@@ -8,104 +8,96 @@ const PageActionsContext = React.createContext({
 })
 PageActionsContext.displayName = 'PageActionsContext'
 
-export class PageActionContainer extends React.PureComponent {
-  render() {
-    return (
-      <PageActionsContext.Consumer>
-        {({ actions }) => actions}
-      </PageActionsContext.Consumer>
-    )
-  }
+export function PageActionContainer() {
+  return (
+    <PageActionsContext.Consumer>
+      {({ actions }) => actions}
+    </PageActionsContext.Consumer>
+  )
 }
 
-export class PageActionProvider extends React.PureComponent {
-  state = {
-    actions: null,
-  }
+export function PageActionProvider(props) {
+  // state = {
+  //   actions: null,
+  // }
+  const [actions, setActions] = useState(null)
 
-  _mountCount = 0
-  _mounted = false
+  let _mountCount = 0
+  let _mounted = false
+  let _pending = null
```

Comment:
I'm not sure that these will work the same in hooks -- each render will re-run this function, re-declaring them, rather than persisting. I'll try to spend some time this afternoon investigating.